sympy/series/limitseq.py

```diff
@@ -193,29 +196,45 @@ def limit_seq(expr, n=None, trials=5):
     .. [1] Computing Limits of Sequences - Manuel Kauers
     """
 
+    from sympy.concrete.summations import Sum
+    from sympy.calculus.util import AccumulationBounds
     if n is None:
         free = expr.free_symbols
         if len(free) == 1:
             n = free.pop()
         elif not free:
             return expr
         else:
-            raise ValueError("expr %s has more than one variables. Please"
-                             "specify a variable." % (expr))
+            raise ValueError("Expression has more than one variable. "
+                             "Please specify a variable.")
     elif n not in expr.free_symbols:
         return expr
 
     expr = expr.rewrite(fibonacci, S.GoldenRatio)
     n_ = Dummy("n", integer=True, positive=True)
+    n1 = Dummy("n", odd=True, positive=True)
+    n2 = Dummy("n", even=True, positive=True)
 
-    # If there is a negative term raised to a power involving n, consider
-    # even and odd n separately.
+    # If there is a negative term raised to a power involving n, or a
+    # trigonometric function, then consider even and odd n separately.
     powers = (p.as_base_exp() for p in expr.atoms(Pow))
-    if any(b.is_negative and e.has(n) for b, e in powers):
-        L1 = _limit_seq(expr.xreplace({n: 2*n_}), n_, trials)
-        if L1 is not None:
-            L2 = _limit_seq(expr.xreplace({n: 2*n_ + 1}), n_, trials)
-            if L1 == L2:
-                return L1
+    if (any(b.is_negative and e.has(n) for b, e in powers) or
+        expr.has(cos, sin)):
+            L1 = _limit_seq(expr.xreplace({n: n1}), n1, trials)
+            if L1 is not None:
+                L2 = _limit_seq(expr.xreplace({n: n2}), n2, trials)
+                if L1 != L2:
+                    if L1.is_comparable and L2.is_comparable:
+                        return AccumulationBounds(Min(L1, L2), Max(L1, L2))
+                    else:
+                        return None
+    else:
+        L1 = _limit_seq(expr.xreplace({n: n_}), n_, trials)
+    if L1 is not None:
+        return L1
     else:
-        return _limit_seq(expr.xreplace({n: n_}), n_, trials)
+        # Maybe the absolute value is easier to deal with (though not if
+        # it's a sum). If it tends to 0, the limit is 0.
+        if not(expr.has(Sum) or expr.is_Add):
```

Comment:
Could sums like `n/(n + 1) + (I/2)**n` be handled by looking at each term separately?